### DISCUSSION TITLE
Always Hard Refresh?

### DISCUSSION THREAD

**Discussion by andrewm-aero** [argoproj/argo-cd#8309] [Q&A] [answered]
We have an application that, due to using date functions in its helm chart, needs to be hard refreshed at midnight every night, and we'd like to be able to have the periodic refreshes be hard refreshes for this app. Is this possible?

I found [this](https://github.com/argoproj/argo-cd/blob/cc4eea0d6951f1025c9ebb487374658186fa8984/pkg/apis/application/v1alpha1/application_annotations.go#L6) which would almost be what we're looking for, but the docs say that this annotations is removed after every refresh, which defeats the purpose. While we could just set a CronJob to patch the app resource every night at midnight, it'd be much nicer if this was just a configuration value we set somewhere.

**Comment by schmel** [comment, marked as answer]
https://github.com/argoproj/argo-cd/issues/4002
https://github.com/argoproj-labs/argocd-vault-plugin/issues/224
